proxy/api/src/http/diagnostics.rs

```diff
@@ -0,0 +1,58 @@
+// Copyright © 2021 The Radicle Upstream Contributors
+//
+// This file is part of radicle-upstream, distributed under the GPLv3
+// with Radicle Linking Exception. For full terms see the included
+// LICENSE file.
+
+//! Provide diagnostics information.
+
+use warp::{filters::BoxedFilter, path, Filter, Rejection, Reply};
+
+use crate::{context, http};
+
+/// Combination of all diagnostics filters.
+pub fn filters(ctx: context::Context) -> BoxedFilter<(impl Reply,)> {
+    // I couldn't figure out how to get past the type checker without having chaining.
+    get_filter(ctx.clone()).or(get_filter(ctx)).boxed()
+}
+
+/// `GET /diagnostics`
+fn get_filter(
+    ctx: context::Context,
+) -> impl Filter<Extract = impl Reply, Error = Rejection> + Clone {
+    path::end()
+        .and(warp::get())
+        .and(http::with_context_unsealed(ctx))
+        .and_then(handler::get)
+}
+
+/// Diagnostics handlers for conversion between core domain and http request fullfilment.
+mod handler {
+    use serde_json::json;
+    use walkdir::WalkDir;
+    use warp::{reply, Rejection, Reply};
+
+    use crate::context;
+
+    /// Get the [`identity::Person`] for the given `id`.
+    #[allow(clippy::unused_async)]
+    pub async fn get(ctx: context::Unsealed) -> Result<impl Reply, Rejection> {
+        let git_dir = ctx.rest.paths.git_dir();
+        let tree = WalkDir::new(git_dir)
+            .into_iter()
+            .filter_map(std::result::Result::ok)
+            .map(|e| {
+                e.path()
+                    .strip_prefix(git_dir)
+                    .expect("Couldn't strip prefix.")
+                    .display()
+                    .to_string()
+            })
+            .collect::<Vec<String>>();
+        let reply = json!({
+            "git_dir": git_dir,
+            "git_dir_tree": tree,
```

Comment:
Can we use camelCase for consistency?

Reply:
94cf1828ed9ef599b9ba55691ef94fc3a82a7aea